bokeh/models/glyphs.py

```diff
@@ -1159,32 +1159,104 @@ class Rect(XYGlyph, LineGlyph, FillGlyph, HatchGlyph):
-
-    The Scatter can draw any built-in marker type. It can be configured
-    to draw the same marker for all values by specifying the name of a
-    marker, e.g.
-
-    .. code-block:: python
-
-        glyph = Scatter(x="x", y="y", size="sizes", marker="square")
-        plot.add_glyph(source, glyph)
-
-    will render only Square markers for all points. Alternatively, the
-    Scatter marker can be configured to use marker types specified in a
-    data source column:
-
-    .. code-block:: python
-
-        # source.data['markers'] = ["circle", "square", "circle", ... ]
-
-        glyph = Scatter(x="x", y="y", size="sizes", marker="markers")
-        plot.add_glyph(source, glyph)
-
-    Note that circles drawn with `Scatter` conform to the standard Marker
-    interface, and can only vary by size (in screen units) and *not* by radius
+    ''' Render scatter markers selected from a predefined list of designs.
+
+    Use Scatter to draw any of Bokeh's built-in marker types:
+    ``asterisk``, ``circle``, ``circle_cross``, ``circle_dot``, ``circle_x``,
+    ``circle_y``, ``cross``, ``dash``, ``diamond``, ``diamond_cross``,
+    ``diamond_dot``, ``dot``, ``hex``, ``hex_dot``, ``inverted_triangle``,
+    ``plus``, ``square``, ``square_cross``, ``square_dot``, ``square_pin``,
+    ``square_x``, ``star``, ``star_dot``, ``triangle``, ``triangle_dot``,
+    ``triangle_pin``, ``x``, or ``y``.
+
+    Bokeh's built-in markers consist of a set of base markers, most of which can
+    be combined with different kinds of additional visual features:
+
+    .. bokeh-plot::
+        :source-position: none
+
+        from bokeh.core.enums import MarkerType
+        from bokeh.io import show
+        from bokeh.plotting import figure
+
+        bases = [
+            "asterisk",
+            "circle",
+            "cross",
+            "dash",
+            "diamond",
+            "dot",
+            "hex",
+            "inverted_triangle",
+            "plus",
+            "square",
+            "star",
+            "triangle",
+            "x",
+            "y",
+        ]
+
+        kinds = [
+            "",
+            "cross",
+            "dot",
+            "pin",
+            "x",
+            "y",
+        ]
+
+        data = []
+
+        for base in bases:
+            for kind in kinds:
+                name = f"{base}_{kind}" if kind else base
+                if name in MarkerType:
+                    data.append((name, base, kind))
+
+        marker, base, kind = zip(*data)
+
+        p = figure(x_range=kinds, y_range=list(reversed(bases)), width=400,
+                toolbar_location=None, x_axis_location="above")
+        p.grid.grid_line_color = None
+        p.axis.major_tick_line_color = None
+
+        p.scatter(
+            marker=marker,
+            x=kind,
+            y=base,
+            size=25,
+            fill_alpha=0.4,
+            fill_color="orange"
+            )
+
+        show(p)
```

Comment:
should move the source code to a file under `sphinx/source/docs/user_guide/examples` (left the inline option for outside users of the extension but generally try to only refer to files in the project docs)